app/Controller/DictionariesController.php

```diff
@@ -40,6 +40,39 @@ class DictionariesController extends StandardController {
       'description' => 'asc'
     )
   );
+
+  /**
+   * Perform any dependency checks required prior to a delete operation.
+   * - postcondition: Session flash message updated (HTML) or HTTP status returned (REST)
+   *
+   * @since  COmanage Registry v4.0.1
+   * @param  Array Current data
+   * @return boolean true if dependency checks succeed, false otherwise.
+   */
+
+  function checkDeleteDependencies($curdata) {
+    // We can't delete a dictionary if it is linked to an attribute enumerator
+
+    if(!empty($curdata["AttributeEnumeration"])) {
+      $deleted_status = Hash::extract($curdata, 'AttributeEnumeration.{n}.deleted');
```

Comment:
As a performance best practice, we should probably be using the various `_contains` settings on the controller. In this case, you could set `$delete_contains` to `array('AttributeEnumeration')`, and I'm not sure if `$view_contains` and `$edit_contains` need anything or can just be set to empty arrays.

Reply:
Done